query-engine/connectors/query-connector/src/query_arguments.rs

```diff
@@ -106,21 +111,40 @@ impl QueryArguments {
 
         // Partition into orderings on the same model and ones that require relation hops.
         // Note: One ordering is always on one scalar in the end.
-        let (on_model, on_relation): (Vec<&OrderBy>, Vec<&OrderBy>) =
-            self.order_by.iter().partition(|o| o.path.is_empty());
+        let (on_model, on_relation): (Vec<&OrderBy>, Vec<&OrderBy>) = self.order_by.iter().partition(|o| match o {
+            OrderBy::Scalar(o) => o.path.is_empty(),
+            OrderBy::Aggregation(o) => o.path.is_empty(),
+            OrderBy::Relevance(_) => true,
+        });
 
         // Indicates whether or not a combination of contained fields is on the source model (we don't check for relations for now).
         let order_by_contains_unique_index = self.model.unique_indexes().into_iter().any(|index| {
-            index
-                .fields()
-                .into_iter()
-                .all(|f| on_model.iter().any(|o| o.field == f))
+            index.fields().into_iter().all(|f| {
+                on_model.iter().any(|o| match o {
+                    OrderBy::Scalar(o) => o.field == f,
+                    OrderBy::Aggregation(_) => false,
+                    OrderBy::Relevance(o) => o.fields.contains(&f),
+                })
+            })
         });
```

Comment:
This needs to be carefully reviewed

Reply:
Just because a field is unique doesn't mean the relevance of that field is unique as well. Needs to return `false`. Same as above, it'd be great to get some tests for those paths.